includes/class-sensei-course.php

```diff
@@ -3751,6 +3768,32 @@ public static function can_current_user_manually_enrol( $course_id ) {
 		return (bool) apply_filters( 'sensei_can_user_manually_enrol', $can_user_manually_enrol, $course_id );
 	}
 
+	/**
+	 * Check if self enrollment is not allowed for the given course.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param int $course_id Course post ID.
+	 *
+	 * @return boolean Whether self enrollment is not allowed.
+	 */
+	public static function is_self_enrollment_not_allowed( $course_id ) {
+		$self_enrollment_not_allowed = (bool) get_post_meta( $course_id, '_self_enrollment_not_allowed', true );
+
+		/**
+		 * Check if self enrollment is not allowed.
+		 *
+		 * @since $$next-version$$
+		 *
+		 * @hook sensei_self_enrollment_not_allowed
+		 *
+		 * @param {bool} $self_enrollment_not_allowed True if self enrollment is not allowed, false otherwise.
+		 * @param {int}  $course_id                   Course post ID.
+		 * @return {bool} Filtered value.
+		 */
+		return (bool) apply_filters( 'sensei_self_enrollment_not_allowed', $self_enrollment_not_allowed, $course_id );
```

Comment:
Just a thought as a potential developer, maybe we can pass the user id here?
In this case, the filter can behave differently depending on the user.
I can imagine scenarios when we might want to allow enrolling for some roles.
Not necessary though. Especially, if we consider it only as a course setting.

Reply:
Hm! Thinking a little about this, I'd be a little concerned about adding the user here because the method and hook are not directly tied to the user, but just a course setting. Currently, I think all the things we're doing are tied to users under the hood, but this could not be true in the future. For example, if we have a cron looping through courses and getting this setting to do something.

And if the users want to make a specific check by the current user in some situation anyway, they could just use the `get_current_user_id()`.

WDYT?

Reply:
Sounds reasonable. Thanks for considering.